systemtest/configure_proxy_stage.py

```diff
@@ -0,0 +1,135 @@
+#!/usr/bin/env python3
+"""Configure system-wide proxy settings when ENV_FOR_DYNACONF is 'stage'.
+
+Requires: SETTINGS_URL env var pointing to a TOML settings file.
+"""
+
+import configparser
+import os
+import subprocess
+import sys
+import urllib.error
+import urllib.request
+from pathlib import Path
+
+try:
+    import tomllib  # Python 3.11+
+except ModuleNotFoundError:
+    import tomli as tomllib
+
+
+ENV_VAR = "ENV_FOR_DYNACONF"
+TARGET_ENV = "stage"
+
+RHSM_CONF = Path("/etc/rhsm/rhsm.conf")
+INSIGHTS_CONF = Path("/etc/insights-client/insights-client.conf")
+YGGDRASIL_OVERRIDE_DIR = Path("/etc/systemd/system/yggdrasil.service.d")
+YGGDRASIL_OVERRIDE = YGGDRASIL_OVERRIDE_DIR / "proxy.conf"
+PROFILE_PROXY = Path("/etc/profile.d/proxy.sh")
+
+
+def load_proxy_settings() -> tuple[str, str, str]:
+    """Download and extract noauth_proxy host/port from [stage] section."""
+    settings_url = os.environ.get("SETTINGS_URL")
+    if not settings_url:
+        raise RuntimeError("SETTINGS_URL must be provided")
```

Comment:
This might break the ability to run tmt locally. We should only check for settings_url if there isn't a local settings.toml file in the top level of the project.